src/blockdata/transaction.rs

```diff
@@ -457,9 +417,11 @@ impl RawExtraField {
     }
 }
 
-impl From<ExtraField> for RawExtraField {
-    fn from(extra: ExtraField) -> Self {
-        crate::consensus::encode::deserialize(&serialize(&extra)).unwrap()
+impl TryFrom<ExtraField> for RawExtraField {
+    type Error = encode::Error;
+
+    fn try_from(extra: ExtraField) -> Result<Self, Self::Error> {
+        encode::deserialize(&serialize(&extra))
```

Comment:
I don't think this can fail, it's just decoding a `Vec<u8>` which will always be correct as we are the one creating the `Vec<u8>`, decoding a `Vec<u8>` can only fail if the VarInt length is wrong or if the amount of data is too large.

Reply:
I changed the function `From<ExtraField> for RawExtraField` to `impl TryFrom<ExtraField> for RawExtraField` because it panicked in one of the coverage tests.

Reply:
Ok, I removed this and replaced it with the `impl From<ExtraField> for RawExtraField` version again because for this PR I could not find the panic again. Please note that `impl From<ExtraField> for RawExtraField` can still cause panics.